spirv_glsl.cpp

```diff
@@ -3221,6 +3239,199 @@ string CompilerGLSL::access_chain(uint32_t base, const uint32_t *indices, uint32
 	return expr;
 }
 
+string CompilerGLSL::access_chain(uint32_t base, const uint32_t *indices, uint32_t count, const SPIRType &target_type)
+{
+	if (flattened_buffer_blocks.count(base))
+	{
+		std::string expr;
+		flattened_access_chain(expr, base, indices, count, target_type, 0);
+
+		return expr;
+	}
+	else
+	{
+		return access_chain(base, indices, count, false);
+	}
+}
+
+void CompilerGLSL::flattened_access_chain(std::string &expr, uint32_t base, const uint32_t *indices, uint32_t count, const SPIRType &target_type, uint32_t offset)
```

Comment:
I was mostly wary of excessive string copies, and I'd need to return smth like a std::pair from flattened_access_chain_offset. I can do that though if this is more consistent with the rest of the code.